chatminer/chatparsers.py

```diff
@@ -227,39 +227,35 @@ def _parse_message(self, mess):
 
 
 class TelegramJsonParser(Parser):
-    def __init__(self, filepath, chat_name="", saved_messages=False):
-        self.saved_messages = saved_messages
+    def __init__(self, filepath, chat_name=""):
         self.chat_name = chat_name
         super().__init__(filepath)
 
     def _read_file_into_list(self):
         self._logger.info("Starting reading raw messages into memory...")
         with self._file.open(encoding="utf-8") as f:
             json_objects = json.load(f)
-        chat_found = False
         if "messages" in json_objects:
             self.messages = json_objects["messages"]
-            chat_found = True
         else:
-            for i, chat in enumerate(json_objects["chats"]["list"]):
-                if chat["type"] == "saved_messages":
-                    if self.saved_messages:
-                        self.messages = json_objects["chats"]["list"][i]["messages"]
-                        chat_found = True
-                        break
-                elif chat["name"] == self.chat_name:
-                    self.messages = json_objects["chats"]["list"][i]["messages"]
-                    chat_found = True
+            if not self.chat_name:
+                self._logger.info(
+                    "No chat name was specified, parsing Saved Messages..."
+                )
+            for chat in json_objects["chats"]["list"]:
+                if ("name" in chat and chat["name"] == self.chat_name) or (
+                    not self.chat_name and chat["type"] == "saved_messages"
+                ):
+                    self.messages = chat["messages"]
```

Comment:
This looks cleaner already, but see my comment in the previous review: I'm not sure if we need to incorporate the `saved_messages` type at all if there's no clear use case for it

Reply:
> This looks cleaner already, but see my comment in the previous review: I'm not sure if we need to incorporate the `saved_messages` type at all if there's no clear use case for it

It's one of the chats, but doesn't have a name, see https://github.com/joweich/chat-miner/pull/47#discussion_r1045097600

Reply:
So it's a collection of messages the user saved across all chats? Do you see an use case for parsing them? I would otherwise just ignore this type to declutter the code, but I'm interested in your opinion

Reply:
> So it's a collection of messages the user saved across all chats? Do you see an use case for parsing them? I would otherwise just ignore this type to declutter the code, but I'm interested in your opinion

If you've never used it or Telegram, It's almost as a chat with yourself (on WhatsApp, I called this a "cloud chat" and it can be done by creating a group with someone and then removing them, so that you're the only one left), and you can use it as some kind of information storage (or at least, that's the use I see), as well as holding documents and such.

As for a use case for parsing said chat, it depends: the code would go from this
```python
if not self.chat_name:
    self._logger.info(
        "No chat name was specified, parsing Saved Messages..."
    )
for chat in json_objects["chats"]["list"]:
    if ("name" in chat and chat["name"] == self.chat_name) or (
        not self.chat_name and chat["type"] == "saved_messages"
    ):
        self.messages = chat["messages"]
```
to this
```python
if not self.chat_name:
    self._logger.error(
        "No chat name was specified."
    )
else:
    for chat in json_objects["chats"]["list"]:
        if "name" in chat and chat["name"] == self.chat_name:
            self.messages = chat["messages"]
```
Meaning a simpler if (and a couple other changes after that, with the logging messages for not finding the chat name specified.

---
Now, thank you for asking my opinion:

> As for a use case for parsing said chat, it depends
It also kind of depends on what is the scope of the `chat-miner`, because one could in theory wanna analyze their saved messages, but that can be done by only exporting that one chat data from Telegram. I would say the one thing the code as is now adds is the "easiness" of choosing whichever chat, even multiple ones (with multiple parsers), without needing to have several different Telegram JSON export files.

The if to check whether the `name` key is there (which I think it isn't only in the case of the Saved Messages chat) has to stay together with the filter for the `chat_name` chosen (unless one uses a different way to explore the JSON object, without linearly exploring every chat), so I would say it's not too much of a problem leaving the option to not specify a chat name and inferring the choice of Saved Messages.

In the end, I defer to your decision, as the way the code works doesn't really change. It would just completely ignore the problematic chat and only pass over the ones that have the `name` key.

Reply:
Wow, so many thanks for this extensive explanation!
I see the rational of the saved messages and also see value in having them in the code.
I would therefore suggest to stick to your logic, but do some minor refactoring to untangle the chained if statement:
```
if self.chat_name:
    for chat in json_objects["chats"]["list"]:
        self._logger.info(f"Seeking for chat {self.chat_name}...")
        if "name" in chat:
            if chat["name"] == self.chat_name:
                self.messages = chat["messages"]
            else:
                self._logger.info(f"Skipping chat {chat['name']}.")
else:
    self._logger.warning("No chat name was specified, parsing Saved Messages...")
    for chat in json_objects["chats"]["list"]:
        if chat["type"] == "saved_messages":
            self.messages = chat["messages"]
        else:
            self._logger.info(f"Skipping chat {chat['name']}.")
```

I can't test this snipped as I don't have an export file with a saved message entry.
I would kindly ask you to do this and commit the result if the test is fine, I will then happily merge this PR!

Reply:
> Wow, so many thanks for this extensive explanation! I see the rational of the saved messages and also see value in having them in the code. I would therefore suggest to stick to your logic, but do some minor refactoring to untangle the chained if statement:
> 
> ```
> if self.chat_name:
>     for chat in json_objects["chats"]["list"]:
>         self._logger.info(f"Seeking for chat {self.chat_name}...")
>         if "name" in chat:
>             if chat["name"] == self.chat_name:
>                 self.messages = chat["messages"]
>             else:
>                 self._logger.info(f"Skipping chat {chat['name']}.")
> else:
>     self._logger.warning("No chat name was specified, parsing Saved Messages...")
>     for chat in json_objects["chats"]["list"]:
>         if chat["type"] == "saved_messages":
>             self.messages = chat["messages"]
>         else:
>             self._logger.info(f"Skipping chat {chat['name']}.")
> ```
> 
> I can't test this snipped as I don't have an export file with a saved message entry. I would kindly ask you to do this and commit the result if the test is fine, I will then happily merge this PR!

Oh damn, you're right, I don's know why for some reason I wanted to keep it all in one loop, it does make much more sense to split it like that. Changes coming right away!

Reply:
I see, good call. I meant to put it in there to give the user some feedback which chats exists if they made a typo when setting the `chat_name`. But feel free to ommit it!